woocommerce/woocommerce
# Shard unit test runs

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Divides our unit tests into two test suites that, after experimenting with different combinations, I've found to have similar runtimes that are also possible to define in a light-weight and relatively future-proof manner. Adds those two shards to the matrix of the CI and PR unit test workflows. 

Closes #38973 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. ensure checks pass
2. ensure the unit test shards were executed as intended
3. I'd especially be interested in whether there's a better way to handle the `include` parts of the matrix -- when not adding a shard explicitly, the `unittests` option is left empty, but copying as in [07db3db](https://github.com/woocommerce/woocommerce/pull/39302/commits/07db3db977512c7e9f2287dc44a8f5148328be49) feels sub-optimal. I found [this issue](https://github.com/orgs/community/discussions/7835) that seems to imply that that's probably the only option? 

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -→


<details>

#### Significance

<!-- Choose only one -→


#### Type

<!-- Choose only one -→


#### Message <!-- Add a changelog message here -→

#### Comment <!-- If the changes in this pull request don't warrant a changelog entry, you can alternatively supply a comment here. Note that comments are only accepted with a significance of "Patch" -→

</details>
